starter-code/routes/auth-routes.js

```diff
@@ -0,0 +1,93 @@
+const express = require("express");
+const authRoutes = express.Router();
+
+const User = require("../models/user");
+const bcrypt         = require("bcrypt");
+const bcryptSalt     = 10;
+
+
+authRoutes.get("/", (req, res, next) => {
+  res.render("home");
+});
+
+
+authRoutes.get("/signup", (req, res, next) => {
+  res.render("auth/signup");
+});
+
+
+authRoutes.post("/signup", (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+  const salt     = bcrypt.genSaltSync(bcryptSalt);
+  const hashPass = bcrypt.hashSync(password, salt);
+
+
+  if (username === "" || password === "") {
+    res.render("auth/signup", {
+      errorMessage: "Indicate a username and a password to sign up"
+    });
+    return;
+  }
+
+
+  User.findOne({ username: username }, "username", (err, user) => {
+    if (user !== null) {
+      res.render("auth/signup", {
+        errorMessage: "Username already exist"
+      });
+      return;
+    }
+
+    const newUser  = User({
+      name: username,
```

Comment:
The User schema doesn't have any `name`, you actually have a `username` field. And this is why the user is not correctly saved in the DB.